Modules/Core/Mesh/include/itkTriangleMeshCurvatureCalculator.h

```diff
@@ -0,0 +1,143 @@
+
+namespace itk
+{
+template <typename TInputMesh>
+class ITK_TEMPLATE_EXPORT TriangleMeshCurvatureCalculator : public Object
+{
+public:
+  ITK_DISALLOW_COPY_AND_MOVE(TriangleMeshCurvatureCalculator);
+
+  /**
+   * Standard class type aliases.
+   */
+  using Self = TriangleMeshCurvatureCalculator;
+
+  /**
+   * Standard "Superclass" type alias.
+   */
+  using Superclass = Object;
+
+  /**
+   * Smart pointer type alias support
+   */
+  using Pointer = SmartPointer<Self>;
+  using ConstPointer = SmartPointer<const Self>;
+
+  /**
+   * Method for creation through the object factory.
+   */
+  itkNewMacro(Self);
+
+  /** Run-time type information (and related methods). */
+  itkTypeMacro(TriangleMeshCurvatureCalculator, Object);
+
+  /**
+   * Convenient type alias for this Calculator.
+   */
+  using InputMeshType = TInputMesh;
+  using MeshConstPointer = typename InputMeshType::ConstPointer;
+  using MeshPointType = typename InputMeshType::PointType;
+  using MeshPointIdConstIterator = typename InputMeshType::CellTraits::PointIdConstIterator;
+  using MeshPointsContainerConstPointer = typename InputMeshType::PointsContainerConstPointer;
+  using MeshPointDataContainer = typename InputMeshType::PointDataContainer;
+  using MeshPointDataContainerPointer = typename InputMeshType::PointDataContainerPointer;
+  using CellsContainerConstPointer = typename InputMeshType::CellsContainerConstPointer;
+  using CellsContainerConstIterator = typename InputMeshType::CellsContainer::ConstIterator;
+  using CellType = typename InputMeshType::CellType;
+
+  /**
+   * Different modes of operation. Use these to specify
+   * how to extract the regions. Currently only Guassian curvature is supported.
+   * Using enum for future extension to Mean, Max and Min curvature.
+   */
+  enum
+  {
+    CurvatureGauss = 0
+  };
+
+  /**
+   * Methods specify mode of operation for the Calculator.
+   */
+  itkSetMacro(CurvatureType, int);
+  itkGetConstMacro(CurvatureType, int);
+
+  itkSetMacro(InvertMeanCurvature, int);
+  itkGetConstMacro(InvertMeanCurvature, int);
+
+  /** Set the input mesh. */
+  itkSetObjectMacro(TriangleMesh, InputMeshType);
+
+  itkGetObjectMacro(GaussCurvatureData, MeshPointDataContainer);
+
+  void
+  SetCurvatureTypeToGaussian()
+  {
+    this->SetCurvatureType(Self::CurvatureGauss);
+  }
+
+  /** Compute the selected curvature type */
+  void
+  Compute();
+
+protected:
+  TriangleMeshCurvatureCalculator();
+  ~TriangleMeshCurvatureCalculator() override = default;
+
+  void
+  PrintSelf(std::ostream & os, Indent indent) const override;
+
+  /**
+   * Discrete Gauss curvature (K) computation
+   */
+  void
+  GetGaussCurvature(MeshConstPointer input);
+
+  void
+  ComputeGaussCurvature(MeshConstPointer input);
+
+private:
+  // Vars
+  int                           m_CurvatureType;
+  bool                          m_InvertMeanCurvature;
+  MeshPointDataContainerPointer m_GaussCurvatureData;
+  MeshConstPointer              m_TriangleMesh;
```

Comment:
If all iVars can be initialized at compile time, the constructor can be defaulted. Less code, and gives compiler some optimization opportunities.